### PR TITLE
17 teaser with image

### LINKED ISSUE #17: Hotel Teaser With Image
implement "image near hotels" block.
- Prominent image with link/text in lower right corner
- Side column with description and location icon

Content so far: 
Screenshot 2023-02-09 at 08 23 09

How it looks on https://www.choicehotels.com/explore/road-trips/best-winter-road-trips

Mobile:
Screenshot 2023-02-09 at 08 23 36

Desktop:
Screenshot 2023-02-09 at 08 23 48

### PR DESCRIPTION
**Pull request by cuzuco2**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #17

Test URLs:
- Before: https://main--choice--hlxsites.hlx.page/en-us/explore/best-winter-road-trips
- After: https://17-teaser-with-image--choice--hlxsites.hlx.page/en-us/explore/best-winter-road-trips

### CODE REVIEW

**Comment by solaris007** [comment]
Some small spacing / position adjustments are needed:
Screenshot 2023-02-14 at 06 35 36

**Comment by solaris007** [comment]
There's a duplication of markup (twice "View hotels ...", one hidden with `display: none`). Why is this needed?
Screenshot 2023-02-14 at 06 39 05

File: blocks/image-near-hotels/image-near-hotels.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  arrowRight.innerHTML = '&nbsp;';
+
+  const textWithArrow = divForLink.innerHTML.trim();
+  const linkWithArrow = textWithArrow.slice(0, (textWithArrow.length) - 4)
+  + arrowRight.outerHTML + textWithArrow.slice((textWithArrow.length) - 4);
+  divForLink.innerHTML = linkWithArrow;
+  const picture = block.querySelectorAll('.image-near-hotels > div > div');
+  picture[0].appendChild(divForLink);
+
+  const mapMarker = document.createElement('span');
```
Comment (solaris007) [inline blocks/image-near-hotels/image-near-hotels.js:22]: i don't think you need an extra span for the map marker - you can prepend it directly to the text by using the before pseudo class:
```
.image-near-hotels.block > div:nth-child(3) > div:before {
    content: '';
    font-family: 'chFont';
}
```

File: blocks/image-near-hotels/image-near-hotels.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const divForLink = document.createElement('div');
+  divForLink.className = 'text-in-image';
+  divForLink.innerHTML = linkInImage[0].firstChild.outerHTML;
+
+  const arrowRight = document.createElement('span');
+  arrowRight.className = 'ch-icon-arrow-right';
+  arrowRight.innerHTML = '&nbsp;';
+
+  const textWithArrow = divForLink.innerHTML.trim();
+  const linkWithArrow = textWithArrow.slice(0, (textWithArrow.length) - 4)
```
Comment (solaris007) [inline blocks/image-near-hotels/image-near-hotels.js:16]: don't need an extra span i think. just directly append arrow with CSS

**Review by solaris007 — COMMENTED** [review]
I think we can drop all of the JS.

File: blocks/image-near-hotels/image-near-hotels.css [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  font-size: var(--font-size-medium);
+  letter-spacing: -0.25px;
+  text-align: left;
+  line-height: 27px;
+}
+
+.image-near-hotels.block > div:nth-child(3) > div {
+  border-top: 1px solid;
+  border-color: var(--article-teaser-border);
+  padding-top: 30px;
```
Comment (solaris007) [inline blocks/image-near-hotels/image-near-hotels.css:22]: use var `--spacing-l`

File: blocks/image-near-hotels/image-near-hotels.css [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+.image-near-hotels.block::after, .image-near-hotels.block::before {
+  content: ' ';
+  display: table;
+}
+
+.ch-icon-map-marker::before {
+  font-family: chFont, sans-serif;
+  content: '';
+  font-style: normal;
+  font-weight: 400;
```
Comment (solaris007) [inline blocks/image-near-hotels/image-near-hotels.css:34]: use var `--font-weight-normal`

File: blocks/image-near-hotels/image-near-hotels.css [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  position: relative;
+  vertical-align: middle;
+}
+
+.ch-icon-arrow-right::before {
+  font-family: chFont, sans-serif;
+  content: '';
+  padding-left: 5px !important;
```
Comment (solaris007) [inline blocks/image-near-hotels/image-near-hotels.css:42]: !important should not be used

File: blocks/image-near-hotels/image-near-hotels.css [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  font-size: 10px !important;
```
Comment (solaris007) [inline blocks/image-near-hotels/image-near-hotels.css:43]: !important should not be used

**Review by solaris007 — APPROVED** [review]
provisionally approved, pending demo tonight and followup PR to convert to non-JS

File: blocks/image-near-hotels/image-near-hotels.css (hunk 3 above)
Comment (cuzuco2) [reply blocks/image-near-hotels/image-near-hotels.css:22]: Done

File: blocks/image-near-hotels/image-near-hotels.css (hunk 5 above)
Comment (cuzuco2) [reply blocks/image-near-hotels/image-near-hotels.css:42]: Done

File: blocks/image-near-hotels/image-near-hotels.css [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+main .text-in-image a{
+  font-size: var(--body-font-size-xs);
+  text-align: right;
+  font-family: var(--body-font-family);
+  letter-spacing: 0;
+  text-decoration: none;
+  border: 0;
+  padding: 0;
+  font-style: normal;
```
[the remark this answers is not in the document]
Comment (cuzuco2) [reply blocks/image-near-hotels/image-near-hotels.css:54]: Done

File: blocks/image-near-hotels/image-near-hotels.css (hunk 6 above)
Comment (cuzuco2) [reply blocks/image-near-hotels/image-near-hotels.css:43]: Done

**Pull request merged**